addons/website_event/models/event_event.py

```diff
@@ -116,28 +141,66 @@ def _compute_is_participating(self):
             registration;
         """
         current_visitor = self.env['website.visitor']._get_visitor_from_request()
-        base_domain = [('event_id', 'in', self.ids), ('state', 'in', ['open', 'done'])]
+        events_in_self = [('event_id', 'in', self.ids)] if self else []
         if self.env.user._is_public() and not current_visitor:
-            events = self.env['event.event']
-        elif self.env.user._is_public():
-            events = self.env['event.registration'].sudo().search(
-                expression.AND([base_domain, [('visitor_id', '=', current_visitor.id)]])
-            ).event_id
+            return self.env['event.event']
```

Comment:
To double-check:

If below records are returned is sudo-mode, could sudo here as well for consistency.

That being said I'm not a big fan of the fact they're returned in sudo mode, it's not very natural and could lead to security issues if people use this method and then chain other calls.